rules/generate_rules.py

```diff
@@ -0,0 +1,71 @@
+import json
+import re
+import sys
+from pathlib import Path
+
+from jinja2 import Environment
+
+rules_src = Path(__file__).parent / "rules.json"
+if not rules_src.exists():
+    # This skip is usefull mostly for CI operations when installing only Python deps
+    print("Skipping rules generation, rule file is missing")
+    sys.exit()
+
+FUZZY_RULES = json.loads((Path(__file__).parent / "rules.json").read_text())[
+    "fuzzyRules"
+]
+
+PY2JS_RULE_RX = re.compile(r"\\(\d)", re.ASCII)
+
+# Do not escape anything, we want to generate code as-is, it won't be interpreted as
+# HTML anyway
+JINJA_ENV = Environment(autoescape=False)  # noqa: S701
+
+js_code_template = """// THIS IS AN AUTOMATICALLY GENERATED FILE, DO NOT MODIFY DIRECTLY
+
+export const fuzzyRules = [
+{% for rule in FUZZY_RULES %}  {
+    match: '{{ rule['match'] }}',
+    replace: '{{ rule['replace'] }}',
+  },
+{% endfor %}
+];
+
+"""
+
+js_parent = Path(__file__).parent.parent / "javascript" / "src"
```

Comment:
I find this more readable… but maybe not windows compat
```suggestion
js_parent = Path(__file__).joinpath("../javascript/src").resolve()
```